addons/website_sale_coupon/models/sale_order.py

```diff
@@ -8,6 +8,26 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _try_pending_coupon(self):
+        if not request:
+            return
+
+        pending_coupon = request.session.get('pending_coupon', {})
+        status = {key: value for key, value in pending_coupon.items() if key == 'promo_code'}
+
+        if self and 'promo_code' in status:
+            coupon_code = status.get('promo_code')
+            error_status = self.env['sale.coupon.apply.code'].apply_coupon(self, coupon_code)
```

Comment:
`error_status` :thinking: ? It is not error returned by apply_coupon right?

Reply:
`apply_coupon` do not only return errors, for instance it may return
```python
return {
                            'generated_coupon': {
                                'reward': coupon.program_id.discount_line_product_id.name,
                                'code': coupon.code,
                            }
                        }
                        ```
                        
So your variable called `error_status` is a wrong name, making the code hard to understand.. since the variable name is wrong

Reply:
Ok I see what you mean. I'm using the same name as was being used in `apply_coupon` before though. Do you want to change them both?

Reply:
It was name like that before because it was only holding errors......
Now that you refactored the code, it is also holding `generated_coupon`